src/shared/path-lookup.c

```diff
@@ -260,7 +260,8 @@ static int acquire_generator_dirs(
                 prefix = tempdir;
         else if (scope == UNIT_FILE_SYSTEM)
                 prefix = "/run/systemd";
-        else if (scope == UNIT_FILE_USER) {
+        else {
+                /* UNIT_FILE_USER */
```

Comment:
I'd be more explicit here actually and add `assert(scope == UNIT_FILE_USER)` as first thing in the else branch... But i guess a comment works too.